core/src/server/access_control/host.rs

```diff
@@ -139,47 +140,31 @@ impl std::ops::Deref for Host {
 	}
 }
 
-/// Specifies if domains should be validated.
-#[derive(Clone, Debug, PartialEq, Eq)]
-pub enum DomainsValidation<T> {
-	/// Allow only domains on the list.
-	AllowOnly(Vec<T>),
-	/// Disable domains validation completely.
-	Disabled,
+/// Policy for validating the `HTTP host header`.
+#[derive(Debug, Clone)]
+pub enum AllowHosts {
+	/// Allow all hosts (no filter).
+	Any,
+	/// Allow only specified hosts.
+	Only(Vec<Host>),
 }
 
-impl<T> From<Option<Vec<T>>> for DomainsValidation<T> {
-	fn from(other: Option<Vec<T>>) -> Self {
-		match other {
-			Some(list) => DomainsValidation::AllowOnly(list),
-			None => DomainsValidation::Disabled,
+impl AllowHosts {
+	/// Verify a host.
+	pub fn verify(&self, value: &str) -> Result<(), Error> {
```

Comment:
Ahh okay nice refactor. So were basically replacing `DomainsValidation`, and `is_host_valid` with this `AllowHosts` impl.